### PR TITLE
Add multiple selection

### PR DESCRIPTION
**Pull request by samiheikki**
More of a prototype. Allowing edits from maintainers.
Connects to https://github.com/vaadin/vaadin-list-box/issues/10

Screenshot 2019-06-30 at 13 34 15

---
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-list-mixin/54)

### CODE REVIEW

File: test/vaadin-list-mixin.html [hunk 1]
```diff
@@ -479,6 +479,86 @@
...
+        });
+
+        const shiftKeyClick = new MouseEvent('click', {
+          bubbles: true,
+          cancelable: true,
+          view: window,
+          shiftKey: true
+        });
+
+        it('should clear selected when multiple=true', () => {
```
Comment (jouni) [inline test/vaadin-list-mixin.html]: Did you see this kind of behavior in some other component, like the native <select>? To me, it seems unnecessary to clear the selection.

Going the other direction, setting multiple=false could retain the first selected item in the set and not clear the whole selection.

**Comment by jouni** [comment]
Is this really something we should add to list-mixin? Multiple selected tabs doesn’t make any sense.

I think I would’ve rather added this only to list-box, so we can avoid having a low-level API (`multipleSelected`) in addition the the more ”options-specific” API that was suggested for list-box (`selectedOptions`).

**Comment by tomivirkki** [comment]
Thanks for the PR. The list-mixin, however, might be a wrong layer for it because then all components extending the mixin would automatically be multi-selectable; `<vaadin-list-box>` should have multi-selection while `<vaadin-tabs>` probably shouldn't.

This could work better as an additional mixin that could then be selectively picked for the components that should be multi-selectable.

**Comment by jouni** [comment]
Wow – talk about timing 😄 Me and Tomi commented about the same thing at the exact same minute.

**Comment by tomivirkki** [comment]
The suggested `selectedOptions` is a read-only property ([in `<select>`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/selectedOptions)) and exposes HTML elements in a property. This is something I think we don't have in any other components; might be problematic for the Flow integration also.

Note that similar [`selectedItems`](https://www.webcomponents.org/element/@polymer/paper-listbox/elements/paper-listbox#property-selectedItems) in `paper-listbox` is also read-only while [`selectedValues`](https://www.webcomponents.org/element/@polymer/paper-listbox/elements/paper-listbox#property-selectedValues) isn't. `selectedValues` as an array of numbers might work well here also.

**Comment by jouni** [comment]
> `selectedValues` as an array of numbers might work well here also.

That’s something I’ve been meaning to write an issue about. In `<vaadin-select>`, you can set use the `value` of an item to select it (e.g. `<vaadin-select value="foobar">` when you have `<vaadin-item value="foobar">`). But that’s not possible with list-box currently – I think that should be fixed.

If that happens, then `selectedValues` should probably support indexes and string values as well.

**Comment by tomivirkki** [comment]
Closing in favour of https://github.com/vaadin/vaadin-list-mixin/pull/56

**Pull request closed without merging**